examples/chef/common/clusters/audio-output/AudioOutputManager.h

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/audio-output-server/audio-output-server.h>
+#include <vector>
+
+class AudioOutputManager : public chip::app::Clusters::AudioOutput::Delegate
+{
+    using OutputInfoType = chip::app::Clusters::AudioOutput::Structs::OutputInfoStruct::Type;
+
+public:
+    AudioOutputManager();
+
+    uint8_t HandleGetCurrentOutput() override;
+    CHIP_ERROR HandleGetOutputList(chip::app::AttributeValueEncoder & aEncoder) override;
+    bool HandleRenameOutput(const uint8_t & index, const chip::CharSpan & name) override;
+    bool HandleSelectOutput(const uint8_t & index) override;
+
+protected:
+    uint8_t mCurrentOutput = 1;
+    std::vector<OutputInfoType> mOutputs;
+    // Magic numbers are here on purpose, please allocate memory
+    static constexpr size_t mNameLenMax = 32;
+    char mOutputName[10][mNameLenMax];
```

Comment:
you are keying off the name storage (fixed size) from a variable-size mOutputs (vector).

This is brittle and hard to manage. How about creating a container that contains a OutputInfoType and the storage that it needs (e.g. a std::string for the name). That way you can never mis-match storage and outputs. You can even choose to not store an OutputInfoType since all you need is the storage. so create a `OutputInfoType GetEncodable() const` on the structure and use it during HandleGetoutputList.

Reply:
I propose:

```cpp
struct OutputData {
   uint8_t index;
   std::string name;
   
   /// Encodable valid for the lifetime of `this` as long as
   /// name does not change (as it points inside name)
   OutputInfoType GetEncodable() const {
       OutputInfoType result;
       result.index = index;
       result.name = chip::CharSpan::fromCharString(name.c_str());
       return result;
   }
};

std::vector<OutputData> mOutputs;
```